esp/services/ws_workunits/ws_workunitsAuditLogs.cpp

```diff
@@ -1406,6 +1406,28 @@ void CWsWorkunitsSoapBindingEx::createAndDownloadWUZAPFile(IEspContext& context,
     request->getParameter("WhereSlow", zapInfoReq.whereSlow);
     request->getParameter("IncludeThorSlaveLog", zapInfoReq.includeThorSlaveLog);
     request->getParameter("ZAPFileName", zapInfoReq.zapFileName);
+
+    StringBuffer sendEmailStr, attachZAPReportToEmailStr, portStr;
+    request->getParameter("SendEmail", sendEmailStr);
+    zapInfoReq.sendEmail = atoi(sendEmailStr.str()) ? true : false;
+    if (zapInfoReq.sendEmail)
+    {
+        request->getParameter("AttachZAPReportToEmail", attachZAPReportToEmailStr);
+        zapInfoReq.attachZAPReportToEmail = atoi(attachZAPReportToEmailStr.str()) ? true : false;
+
+        request->getParameter("EmailSender", zapInfoReq.emailSender);
+        request->getParameter("EmailSubject", zapInfoReq.emailSubject);
+        request->getParameter("EmailBody", zapInfoReq.emailBody);
+        //Read maxAttachmentSize from setting.
+        zapInfoReq.maxAttachmentSize = wswService->zapEmailMaxAttachmentSize;
+        zapInfoReq.emailServer.set(wswService->zapEmailServer.get());
+        zapInfoReq.port = wswService->zapEmailServerPort;
+        zapInfoReq.emailTo.set(wswService->zapEmailTo.str());
+
+        if (zapInfoReq.emailSender.isEmpty())
+            zapInfoReq.emailSender.set(zapInfoReq.emailTo.str());
```

Comment:
This looks wrong.

Reply:
Should I throw an exception if zapInfoReq.emailSender is empty?

Reply:
It's the mixture of emailSender and emailTo that I am concerned about

Reply:
How about add a preconfigured emailSender and allow a user to override?